src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-units/attachment-units.component.ts

```diff
@@ -117,7 +117,7 @@ export class AttachmentUnitsComponent implements OnInit {
             )
             .subscribe({
                 next: (res) => {
-                    this.units = res.body!.units;
+                    this.units = res.body?.units || this.units;
                     this.numberOfPages = res.body!.numberOfPages;
```

Comment:
Should this line be changed as well?

Reply:
You mean "?" operator. No I think it should be "!". But the rest should be fine

Reply:
I meant this:
```suggestion
                    this.numberOfPages = res.body!.numberOfPages || this.numberOfPages;
```
or rather this to also handle the case `res.body!.numberOfPages = 0` correctly
```suggestion
                    this.numberOfPages = res.body!.numberOfPages ?? this.numberOfPages;
```

I'm not too familiar with the code but I saw this really similar line and was unsure if we have to change it as well.

Reply:
This is the amount of total page number in the file, wouldn't be always defined? But I can add it for extra security

Reply:
In that case it's fine 👍